WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -477,6 +481,7 @@ public boolean onPrepareOptionsMenu(Menu menu) {
                         } else {
                             saveMenuItem.setTitle(R.string.update_verb);
                         }
+                        mSaveWithoutLeaving = true;
```

Comment:
I agree we shouldn't get this behavior when the button says "Publish". This button always push something to the server (publish a new post, update a published post, update a saved draft, etc.) so this feature feels a bit weird to me.

I guess this is just a bandaid before we implement an auto sync feature. I'm not even sure we should implement this, but if we do I think we should only do it when the button says "Save" (local draft -> draft) or "Upgrade" (draft -> draft, published post -> published post). In any case we'll get the upload notification which makes the experience even weirder.

Ask a designer?